modules/core/arrow-data/src/main/kotlin/arrow/data/Moore.kt

```diff
@@ -0,0 +1,17 @@
+package arrow.data
+
+import arrow.higherkind
+
+@higherkind
+data class Moore<E, V>(val view: V, val handle: (E) -> Moore<E, V>) : MooreOf<E, V> {
```

Comment:
Can you please check if MooreOfJ is available?

Reply:
There is a `MooreKindedJ` if that's what you mean. `StoreKindedJ` and `SumKindedJ` exist too.

Reply:
Yeah those I meant. Make the datatypes implement them!